website/content/docs/concepts/tune-server-performance.mdx

```diff
@@ -0,0 +1,948 @@
+kernel: [16909.874486] Killed process 10742 (consul) total-vm:242812kB, anon-rss:142081kB, file-rss:68768kB
+```
+
+</CodeBlockConfig>
+
+
+Reduced IOPS on the Consul servers is another common cause of issues. This condition can manifest itself in Vault as errors related to canceled context, such as in the following examples.
+
+<CodeBlockConfig hideClipboard>
+
+
+```plaintext
+[ERROR] core: failed to create token: error="failed to persist entry: context canceled"
+
+[ERROR] core: failed to register token lease: request_path=auth/approle/login error="failed to persist lease entry: context canceled"
+
+[ERROR] core: failed to create token: error="failed to persist accessor index entry: context canceled"
+```
+
+</CodeBlockConfig>
+
+
+The key clue here is the "context canceled" message. This issue will cause intermittent Vault availability to all users, and you should try to remedy the issue by increasing the available IOPS for the Consul servers.
+
+---
+
+The following are some important performance related configuration settings that you should become aware of when using Consul for Vault storage.
+
+#### kv_max_value_size
+
+One common performance constraint that you can find when using Consul for Vault storage is the size of data Vault can write as a value to one key in the Consul key/value store.
+
+As of Consul [version 1.7.2](https://github.com/hashicorp/consul/blob/master/CHANGELOG.md#172-march-16-2020) you can explicitly specify this value in bytes with the configuration parameter [kv_max_value_size](/consul/docs/agent/config/config-files#kv_max_value_size).
+
+Default value: **512KB**
+
+Here is an example Consul server configuration snippet that increases this value to **1024KB**.
+
+<CodeBlockConfig hideClipboard>
+
+
+```plaintext
+ "limits": {
+     "kv_max_value_size": 1024000
+ }
+```
+
+</CodeBlockConfig>
+
+
+**What are common errors?**
+
+Vault returns the following error to a client that attempts to exceed the maximum value size.
+
+<CodeBlockConfig hideClipboard>
+
+
+```plaintext
+Error writing data to kv/data/foo: Error making API request.
+
+URL: PUT http://127.0.0.1:8200/v1/kv/data/foo
+Code: 413. Errors:
+
+* failed to parse JSON input: http: request body too large
+```
+
+</CodeBlockConfig>
+
+
+<Note>
+ 
+ Tuning this improperly can cause Consul to fail in unexpected ways, it may potentially affect leadership stability and prevent timely heartbeat signals by increasing RPC IO duration.
+
+</Note>
+
+#### txn_max_req_len
+
+This parameter configures the maximum number of bytes for a transaction request body to the Consul `/v1/txn` endpoint. In situations where you set both `txn_max_req_len` and `kv_max_value_size`, the higher value takes precedence for both settings.
+
+<Note>
+
+Tuning this improperly can cause Consul to fail in unexpected ways, it may potentially affect leadership stability and prevent timely heartbeat signals by increasing RPC IO duration.
```

Comment:
```suggestion
Tuning this improperly can cause Consul to fail in unexpected ways, it may potentially affect leadership stability and prevent regular heartbeat signals by increasing RPC IO duration.
```